source/common/router/config_impl.cc

```diff
@@ -661,16 +672,47 @@ std::string RouteEntryImplBase::newPath(const Http::RequestHeaderMap& headers) c
     final_host = processRequestHost(headers, final_scheme, final_port);
   }
 
-  if (!path_redirect_.empty()) {
-    final_path = path_redirect_.c_str();
-  } else {
-    final_path = headers.getPathValue();
-    if (strip_query_) {
-      size_t path_end = final_path.find("?");
+  std::string final_path_value;
+  if (Runtime::runtimeFeatureEnabled(
+          "envoy.reloadable_features.preserve_query_string_in_redirects")) {
```

Comment:
Is it possible to not check this twice since you check it during route load? It might not be consistent then. Can we latch the expected behavior in the constructor?